src/explorer/model/iDiagnostic.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+import * as fse from "fs-extra";
+import * as vscode from "vscode";
+import { UserError } from "../../utils/errorUtils";
+import { ElementNode, getNodesByTag, XmlTagName } from "../../utils/lexerUtils";
+import { Dependency } from "./Dependency";
+
+export const MAVEN_DEPENDENCY_CONFLICT = "Maven dependency conflict";
+
+class IDiagnostic {
```

Comment:
IDiagnostic indicates it's an interface (but it's not). It looks more like a container of DiagnosticColleciton + several utilities.

Reply:
And it's not part of Maven Explorer. Suggest to rename it a `DiagnositcsProvider`, and move it directly under 'src' folder.